builder/dockerfile/shell_parser.go

```diff
@@ -15,18 +9,26 @@ import (
 	"github.com/pkg/errors"
 )
 
-type shellWord struct {
-	word        string
-	scanner     scanner.Scanner
-	envs        []string
-	pos         int
+// ShellLex performs shell word splitting and variable expansion.
+//
+// ShellLex takes a string and an array of env variables and
+// process all quotes (" and ') as well as $xxx and ${xxx} env variable
+// tokens.  Tries to mimic bash shell process.
+// It doesn't support all flavors of ${xx:...} formats but new ones can
+// be added by adding code to the "special ${} format processing" section
+type ShellLex struct {
```

Comment:
I guess it could be. I think that would imply that this struct was just a rune with some extra behaviour, but I don't think that's really the case.

I'd like to keep it as a struct for two reasons:
* I don't think `rune(shlex)` makes much sense, that should be an error
* the struct is more about it's functionality (shell word and variable expansion), it just happens to only take a single `rune` argument for now.  It's quite possible it will include other fields in the future